src/Json/Value.php

```diff
@@ -606,6 +606,49 @@ public function ¿dateTime(string $format = DateTime::ATOM): ?DateTimeImmutable
         return $this->dateTime($format);
     }
 
+    /**
+     * Assert this value is an int, assert it conforms to a given date time format, then return that date time.
+     * For timestamp strings, use dateTime('U').
+     *
+     * @return DateTimeImmutable
+     * @throws Exception\AssertionFailed If this value is not a string or int, or does not conform to the given format.
+     */
+    public function timestamp(): DateTimeImmutable
+    {
+        $string = (string)$this->int();
+        $result = DateTimeImmutable::createFromFormat('U', $string);
```

Comment:
```
$int = $this->int();

return new DateTimeImmutable("@$int");
```
This can't fail, and if it does, the failure is propagated from the constructor call.